serve/package.go

```diff
@@ -105,13 +105,17 @@ func (s *PluginServe) writeTablesJSON(ctx context.Context, dir string) error {
 func (s *PluginServe) build(pluginDirectory, goos, goarch, distPath, pluginVersion string) (*TargetBuild, error) {
 	pluginName := fmt.Sprintf("plugin-%s-%s-%s-%s", s.plugin.Name(), pluginVersion, goos, goarch)
 	pluginPath := path.Join(distPath, pluginName)
-	args := []string{"build", "-o", pluginPath}
 	importPath, err := s.getModuleName(pluginDirectory)
 	if err != nil {
 		return nil, err
 	}
+	ldFlags := fmt.Sprintf("-s -w -X %s/plugin.Version=%s", importPath, pluginVersion)
+	if getEnvOrDefault("STATIC_LINKING_ENABLED", "0") == "1" {
```

Comment:
Can we make this into a plugin option? See example https://github.com/cloudquery/cloudquery/blob/9e916090240a33ddf5a0163b950bf8fc1a8873ed/plugins/destination/sqlite/main.go#L23

Not sure if we want `withStaticLinking` bool option or `withAdditionalLDFlags` `[]string` flag. Maybe `withStaticLinking` is good for now